src/main/groovy/com/calgaryscientific/gradle/VeracodeDetailedReportCSVTask.groovy

```diff
@@ -26,57 +26,232 @@
         dependsOn 'veracodeDetailedReport'
+        build_id = project.findProperty("build_id")
+        flawsDetailedReportCSVFile = new File("${project.buildDir}/veracode", "detailed-report-flaws-${build_id}.csv")
+        openFlawsDetailedReportCSVFile = new File("${project.buildDir}/veracode", "detailed-report-open-flaws-${build_id}.csv")
+        softwareCompositionAnalysisFile = new File("${project.buildDir}/veracode", "detailed-report-software-composition-analysis-${build_id}.csv")
     }
 
-    void run() {
-        File csvFile = new File('build/scan-results.csv')
-        csvFile.newWriter()
-        csvFile << ["Issue Id",
-                    "Severity",
-                    "Exploit Level",
-                    "CWE Id",
-                    "CWE Name",
-                    "Module",
-                    "Source",
-                    "Source File Path",
-                    "Line",
-                    "Remediation Status",
-                    "Mitigation Status",
-                    "Mitigation Action",
-                    "Mitigation Description",
-                    "Mitigation Date"].join(",") + "\n"
+    VeracodeDetailedReportTask veracodeDetailedReport = new VeracodeDetailedReportTask()
+    private File inputFile = veracodeDetailedReport.getOutputFile()
+
+    // Scan results are not available until the full scan is complete so there is no risk in caching the report.
+    @InputFile
+    File getInputFile() {
+        return inputFile
+    }
+
+    // Scan results are not available until the full scan is complete so there is no risk in caching the report.
+    @OutputFiles
+    List<File> getOutputFiles() {
+        return [flawsDetailedReportCSVFile, openFlawsDetailedReportCSVFile, softwareCompositionAnalysisFile]
+    }
+
+    /**
+     * Extracts the software_composition_analysis section of the detailed XML report and prints it as a separate csv file.
+     * @param xml - detailed report.
+     * @param file - file to write the report to.
+     */
+    static void softwareCompositionAnalysisCSV(Node xml, File file) {
+        BufferedWriter writer = new BufferedWriter(file.newWriter())
+        CSVPrinter csv = new CSVPrinter(writer, CSVFormat.DEFAULT)
+        List<String> headerRow = ['library', 'file_name', 'vendor', 'description', 'cve_id', 'cwe_id', 'cvss_score', 'severity', 'cve_summary']
+        csv.printRecord(headerRow)
+        NodeList componentList = XMLIO.getNodeList(xml, 'software_composition_analysis', 'vulnerable_components', 'component')
+        /**
+         * Component tree:
+         *
+         * /component @description @file_name @library @max_cvss_score @sha1 @vendor @version @vulnerabilities
+         *  /file_paths
+         *      /file_path @value
+         *  /vulnerabilities
+         *      /vulnerability @cve_id @cve_summary @cvss_score @cwe_id @severity
+         *  /violated_policy_rules
+         */
+        for (int i = 0; i < componentList.size(); i++) {
```

Comment:
you could use groovy idiom here:

```
componentList.each { component ->

}
```

Reply:
Groovy idioms don't work when using static compilation.